### PR TITLE
[DataView] Refactor API - Internal refactor

### PR DESCRIPTION
**Pull request by manmorjim**
### Internal changes

- Removed flag `useViewport` from constructor. Set when the filter `viewport` is applied.
- Implemented the method `setSpatialFilter` for `DataViewLocal`.
- Added the method `getFeatures` in layers which returns all the features. For MVT layers it will return just viewport features.
- Added the method `getFeatures` in sources.

### CODE REVIEW

File: src/lib/viz/dataview/DataView.ts
Comment (VictorVelarde) [inline src/lib/viz/dataview/DataView.ts:47]: Nice removal!

File: src/lib/viz/dataview/mode/DataViewLocal.ts
Comment (VictorVelarde) [inline src/lib/viz/dataview/mode/DataViewLocal.ts:59]: I'm ok with this, but I guess that in the future we might need the whole spatialFilter stored

File: src/lib/viz/layer/Layer.ts
Comment (VictorVelarde) [inline src/lib/viz/layer/Layer.ts:302]: @manmorjim I'm not sure if I like the idea of having a common & public `getFeatures` here, with 2 different meanings... And we also are going to keep the `getViewportFeatures`, so it's also difficult to explain for the final user.

I think we should probably remove its use here in the Layer, and keep it in the `Source` where it belongs (IMO this is more a shortcut for our internal use than a clear feature for the user). We should review its external use (I believe it is just in the Dataview internals...) to see if we can avoid it.

I don't want to block this PR, cause I think other changes are good to go, so if you want to discuss more about this, we would merge this and then we will have a bit more of discussion

**Review by VictorVelarde — APPROVED** [review]
OK. I have doubts about the 'getFeatures' on Layer, but feel free to merge and then discuss

File: src/lib/viz/layer/Layer.ts
Comment (manmorjim) [reply src/lib/viz/layer/Layer.ts:302]: Yup, I'd like discussing more about you suggested because I agree with you about it is not a clear from a user point of view having two different methods `getViewportFeatures` and `getFeatures`.
But in the internal use is quite weird the casuistry:
- Layers with `SQLSource` or `DatasetSource` use `getViewportFeatures`.
- Rest of layers use `getFeatures` from their sources.

As you said, let's merge this and having a bit more of discussion.

Comment (VictorVelarde) [reply src/lib/viz/layer/Layer.ts:302]: Agree. Let's merge. 

My 2 cts for the next discussion: right now we can't get the 'whole features' (that's what I understand with 'getFeatures') for tiled layers (maybe in the feature, with a paginated query to SQL, for example).... so...

I see no issue in having only what we can support always: the getViewportFeatures, and leave the 'getFeatures' to those sources which can support it

Comment (manmorjim) [reply src/lib/viz/layer/Layer.ts:302]: Ticket created in CH for the discussion: https://app.clubhouse.io/cartoteam/story/93117/discussion-merge-methods-getfeature-and-getviewportfeatures

**Pull request merged**
